File: 01.JS-Basic/WHILE цикъл/Graduation.js
function graduation(input) {

    let studentName = input[0]
    let i = 1
    let gradeSum = 0
    let fail = 0
    let grade = 0

    while (i < input.length) {
        
        let currentRating = Number(input[i])
        gradeSum += currentRating

        if (currentRating < 4) {
            fail++
            if (fail > 1) {
                console.log(`${studentName} has been excluded at ${grade} grade`)
                break;
            }
        }
        
        grade++
        i++
    }
    if (fail <= 1) {
        console.log(`${studentName} graduated. Average grade: ${(gradeSum / 12).toFixed(2)}`)
    }
}
graduation(["Gosho", 
"5",
"5.5",
"6",
"5.43",
"5.5",
"6",
"5.55",
"5",
"6",
"6",
"5.43",
"5"])

File: JAVA-Basic/JavaBasic/src/WhileLoop/Graduation.java
package WhileLoop;

import java.util.Scanner;

public class Graduation {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        String name = scan.nextLine();
        int classCounter = 1;
        double gradeSum = 0.0;
        int falseCounter = 0;

        while (classCounter <= 12 && falseCounter < 2) {
            double currentGrade = Double.parseDouble(scan.nextLine());

            if (currentGrade < 4.00) {
                falseCounter++;
            } else {
                gradeSum += currentGrade;
                classCounter++;
            }
        }
        if (falseCounter >= 2) {
            System.out.printf("%s has been excluded at %d grade", name, classCounter);
        } else {
            System.out.printf("%s graduated. Average grade: %.2f", name, gradeSum / 12);
        }
    }
}
